src/AtelierTomato.Markov.Service.Discord/DiscordObjectOIDBuilder.cs

```diff
@@ -27,7 +28,7 @@ public static async Task<DiscordObjectOID> Build(IGuild? guild, IChannel channel
 			}
 			else if (channel is IThreadChannel threadChannel)
 			{
-				if (guild is not null && await guild.GetChannelAsync(threadChannel.CategoryId!.Value) is not null and INestedChannel parentChannel && parentChannel.CategoryId is not null)
+				if (await guild.GetChannelAsync(threadChannel.CategoryId!.Value) is not null and INestedChannel parentChannel && parentChannel.CategoryId is not null)

```

Comment:
put the && parentchannel ... thing in front to save unnecessary API calls.
also i would bail out early if we somehow have a thread without a parent and log it as something really bad rather than having a nullreferenceexception in the logs somewhere because of an exclamationpointed null value failing